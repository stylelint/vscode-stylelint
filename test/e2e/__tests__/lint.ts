import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('Linting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should lint CSS documents', async () => {
		const document = await openDocument('defaults/lint.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'color-hex-length',
				codeDescription: 'https://stylelint.io/user-guide/rules/color-hex-length',
				message: 'Expected "#fff" to be "#ffffff" (color-hex-length)',
				range: [2, 9, 2, 13],
				severity: 'error',
			},
		]);
	});

	it('should display rule documentation links when one is available', async () => {
		const document = await openDocument('defaults/rule-doc.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'plugin/foo-bar',
				message: 'Bar (plugin/foo-bar)',
				range: [0, 5, 0, 6],
				severity: 'error',
			},
			{
				code: 'color-hex-length',
				codeDescription: 'https://stylelint.io/user-guide/rules/color-hex-length',
				message: 'Expected "#fff" to be "#ffffff" (color-hex-length)',
				range: [2, 9, 2, 13],
				severity: 'error',
			},
			{
				code: 'color-no-invalid-hex',
				codeDescription: 'https://stylelint.io/user-guide/rules/color-no-invalid-hex',
				message: 'Unexpected invalid hex color "#y3" (color-no-invalid-hex)',
				range: [6, 11, 6, 14],
				severity: 'error',
			},
		]);
	});
});
