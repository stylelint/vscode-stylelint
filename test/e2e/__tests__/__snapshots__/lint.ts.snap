// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linting should display rule documentation links when one is available 1`] = `
[
  {
    "code": "plugin/foo-bar",
    "message": "Bar (plugin/foo-bar)",
    "range": {
      "end": {
        "character": 6,
        "line": 0,
      },
      "start": {
        "character": 5,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "color-no-invalid-hex",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/color-no-invalid-hex",
    },
    "message": "Unexpected invalid hex color "#y3" (color-no-invalid-hex)",
    "range": {
      "end": {
        "character": 14,
        "line": 6,
      },
      "start": {
        "character": 11,
        "line": 6,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "function-name-case",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/function-name-case",
    },
    "message": "Expected "RGB" to be "rgb" (function-name-case)",
    "range": {
      "end": {
        "character": 12,
        "line": 2,
      },
      "start": {
        "character": 9,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;

exports[`Linting should lint CSS documents 1`] = `
[
  {
    "code": "function-name-case",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/function-name-case",
    },
    "message": "Expected "RGB" to be "rgb" (function-name-case)",
    "range": {
      "end": {
        "character": 12,
        "line": 2,
      },
      "start": {
        "character": 9,
        "line": 2,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
